### DISCUSSION TITLE
Return separate destination IP based on some condition

### DISCUSSION THREAD

**Discussion by ihgann** [coredns/coredns#5972] [Q&A]
Hi folks,

I'm trying to understand if/what I would need to change to resolve the following usecase I have.

Based on the source IP, I want to conditionally return a different destination IP when a DNS lookup is performed.

For example, if I detect my source IP is a peer (say a local IP in range `10.24.0.0/29`), I want `ihgann.com` to resolve to a local IP (say `10.24.0.1`), but for all other DNS lookups for this domain, forward to the upstream DNS server.

I am not sure if there are any specific plugins available, or if I write my own plugin(s), if this would be achievable. Has anybody looked into this sort of thing?

I think https://github.com/coredns/coredns/discussions/4940 somewhat touches on this issue, but I'm specifically trying to see if I can return with a resolved IP instead of communicating to a separate DNS server, here. I'm definitely new to CoreDNS, so I'm hoping to determine if there are any upstream ideas that could help, here. Thanks!

**Comment by chrisohaver** [comment]
You can use the _view_ plugin to create a separate view for a particular source IP.  It's a fairly recent feature, added in 1.10.0.

https://github.com/coredns/coredns/tree/master/plugin/view

something along the lines of the following (hand written, untested) corefile ...


```
ihgann.com in-addr.arpa {
  view myview {
    expr incidr(client_ip(), '10.24.0.0/29')
  }
  hosts {
    10.24.0.1 ihgann.com
    fallthrough
  }
  forward . 8.8.8.8
}

. {
  forward . 8.8.8.8
}
```

(edit1: added example)
(edit2: added reverse domain)

**Reply by ihgann** [reply]
Wow, thank you so much for the speedy reply. I had noticed https://coredns.io/plugins/ doesn't list `view` (as it lists up to `v1.9.3`), so maybe that's partially why I wasn't even close to finding it!

I'll test this locally soon and see if it satisfies the usecase, but your example looks like it would do almost exactly what I need. Thank you so much!
